sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -416,6 +419,194 @@ def __call__(
             n_iteration - 1,  # substract 1 to conform with vanilla sklearn count
         )
 
+    def get_labels(
+        self,
+        X,
+        sample_weight,
+        x_squared_norms,
+        centers,
+        n_threads=1,
+        return_inertia=True,
+    ):
+        """This call is expected to accept the same inputs than sklearn's private
+        _labels_inertia and returns (labels, None)
+        """
+        (
+            X,
+            centers,
+            dtype,
+            work_group_size,
+            n_features,
+            n_samples,
+            n_clusters,
+        ) = self._check_inputs(X, centers)
+
+        # Create a set of kernels
+        label_assignment_fixed_window_kernel = make_label_assignment_fixed_window_kernel(
+            n_samples,
+            n_features,
+            n_clusters,
+            preferred_work_group_size_multiple=self.preferred_work_group_size_multiple,
+            centroids_window_width_multiplier=self.centroids_window_width_multiplier,
+            centroids_window_height=self.centroids_window_height,
+            work_group_size=work_group_size,
+            dtype=dtype,
+        )
+
+        half_l2_norm = make_half_l2_norm_2d_axis0_kernel(
+            size0=n_features,
+            size1=n_clusters,
+            work_group_size=work_group_size,
+            dtype=dtype,
+        )
+
+        X_t, centroids_t = self._load_transposed_data_to_device(X, centers)
+
+        centroids_half_l2_norm = dpctl.tensor.empty(
+            n_clusters, dtype=dtype, device=self.device
+        )
+        assignments_idx = dpctl.tensor.empty(
+            n_samples, dtype=np.uint32, device=self.device
+        )
+
+        half_l2_norm(centroids_t, centroids_half_l2_norm)
```

Comment:
This adds a `_kernel` suffix to `half_l2_norm` to match the kernels' naming convention.
```suggestion
        half_l2_norm_kernel = make_half_l2_norm_2d_axis0_kernel(
            size0=n_features,
            size1=n_clusters,
            work_group_size=work_group_size,
            dtype=dtype,
        )

        X_t, centroids_t = self._load_transposed_data_to_device(X, centers)

        centroids_half_l2_norm = dpctl.tensor.empty(
            n_clusters, dtype=dtype, device=self.device
        )
        assignments_idx = dpctl.tensor.empty(
            n_samples, dtype=np.uint32, device=self.device
        )

        half_l2_norm_kernel(centroids_t, centroids_half_l2_norm)
```